server.js

```diff
@@ -0,0 +1,118 @@
+const express = require('express');
+const db = require('./data/db.js');
+const cors = require('cors');
+
+const port = 5555;
+const server = express();
+server.use(express.json());
+server.use(cors({ origin: 'http://localhost:3000' }));
+
+
+server.post('/api/users', (req, res) => {
+  const { name, bio } = req.body;
+  if (!name || !bio) {
+    res.status(400);
+    res.json({ errorMessage: "Please provide name and bio for the user." });
+  }
+  else {
+    db
+      .insert({ name, bio })
+      .then(response => {
+        res.status(201);
+        db.findById(response.id)
+          .then(user => {
+            res.json({ user });
+          });
+      })
+      .catch(error => {
+        res.status(500);
+        res.json({ errorMessage: "There was an error while saving the user to the database" });
+      });
+  }
+});
+
+
+server.get('/api/users', (req, res) => {
+  db.find().then(users => {
+    res.json({ users });
+  })
+  .catch(error => {
+    res.status(500);
+    res.json({ errorMessage: "The users information could not be retrieved." });
+  });
+});
+
+server.get('/api/users/:id', (req, res) => {
+  const id = req.params.id;
+  db
+    .findById(id)
+    .then(users => {
+      if (users.length) {
+        res.status(200);
```

Comment:
This is an interesting approach. What's your logic in using users.length as a conditional?

Reply:
Only just now looking at my comments.  If I recall correctly, I discovered that findById was returning an empty array when nothing was found, which makes users.length 0, which evaluates as false.  When a user was found, it was returning an array containing that user, which would have length 1, evaluating as true.  So users.length could be used just like it was a boolean.  It makes my math-brain happy to see equivalencies like that.